sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -980,9 +988,15 @@ static gcoap_request_memo_t* _find_req_memo_by_token(const sock_udp_ep_t *remote
                 DEBUG("matching multicast response\n");
             }
             else {
+#if SOCK_HAS_IPV4
+                DEBUG("Remote address mismatch %s\n",
+                      ipv4_addr_to_str(_ipv6_addr_str, (ipv4_addr_t *)&memo->remote_ep.addr.ipv4,
+                                       IPV6_ADDR_MAX_STR_LEN));
+#else
                 DEBUG("Remote address mismatch %s\n",
                       ipv6_addr_to_str(_ipv6_addr_str, (ipv6_addr_t *)&memo->remote_ep.addr.ipv6,
                                        IPV6_ADDR_MAX_STR_LEN));
+#endif
                 continue;
```

Comment:
Couldn't we keep the preprocessor-if-blocks short and avoid code duplication by instead doing the following (especially above)?

```suggestion
#if SOCK_HAS_IPV4
                ipv4_addr_to_str(_ipv6_addr_str, (ipv4_addr_t *)&memo->remote_ep.addr.ipv4,
                                       IPV6_ADDR_MAX_STR_LEN);
#else
                ipv6_addr_to_str(_ipv6_addr_str, (ipv6_addr_t *)&memo->remote_ep.addr.ipv6,
                                       IPV6_ADDR_MAX_STR_LEN);
#endif
                DEBUG("Remote address mismatch %s\n", _ipv6_addr_str);
                continue;
```

Reply:
Hum I see, and guarding it with another #if is probably not too nice either :/

Reply:
Can we get the quick fix in for the release?